automated/linux/deqp-runner/deqp-runner.yaml

```diff
@@ -0,0 +1,48 @@
+metadata:
+    format: "Lava-Test Test Definition 1.0"
+    name: deqp-runner
+    description: "The drawElements Quality Program (deqp) Testing allows to test OpenGL API's, uses
+                  OpenGL ES CTS from https://github.com/KhronosGroup/VK-GL-CTS."
+
+    maintainer:
+        - anibal.limon@linaro.org
+    os:
+        - openembedded
+    scope:
+        - functional
+    devices:
+        - dragonboard410c
+        - dragonboard820c
+        - dragonboard845c
+
+params:
+    DISPLAY: ":0"
+    EGL_PLATFORM: "surfaceless"
+    DEQP_BIN: "deqp-gles2"
+    DEQP_CASES: ""
+    DEQP_FAIL: ""
+    DEQP_EXCLUDE: ""
+    DEQP_RUNNER_OPTIONS: "--compact-display false --shuffle false --allow-flakes true"
+    DEQP_RUNNER_JOBS: ""
+    DEQP_OPTIONS: "--deqp-surface-width=256 --deqp-surface-height=256 --deqp-surface-type=pbuffer --deqp-gl-config-name=rgba8888d24s8ms0 --deqp-visibility=hidden --deqp-shadercache=disable"
+
+run:
+    steps:
+        - export DISPLAY=${DISPLAY}
+        - export EGL_PLATFORM=${EGL_PLATFORM}
+        - export RESULT_FILE=result.txt
+        - cd ./automated/linux/deqp-runner
+        - set +e
+        - deqp-runner --deqp ${DEQP_BIN} --output $RESULT_FILE ${DEQP_CASES} ${DEQP_FAIL} ${DEQP_EXCLUDE} ${DEQP_RUNNER_OPTIONS} ${DEQP_RUNNER_JOBS} -- ${DEQP_OPTIONS}
+        - DEQP_EXITCODE=$?
```

Comment:
the whole block below can be replaced with ../../utils/send-to-lava.sh ${RESULT_FILE}. It would also be better to convert the code above this comment to a shell script so it can be executed without yaml file (meaning without LAVA or test-runner.py).

Reply:
@mwasilew I can create a shell script to execute the test, regarding the send-to-lava.sh change the results python script has different regex [1] and has a mapping inside for results [2].

[1] https://github.com/Linaro/test-definitions/pull/209/commits/290d4afdf2bfe9730356cbc35c8755063140f7db#diff-49ed82b07280d7f16b68a1c465b0a882R40
[2] https://github.com/Linaro/test-definitions/pull/209/commits/290d4afdf2bfe9730356cbc35c8755063140f7db#diff-49ed82b07280d7f16b68a1c465b0a882R28

Reply:
@alimon I would prefer to have the RESULT_FILE in a standardized format. If you need to translate results, please save the intermediate file with a different name. Please check iozone.sh as example.

Reply:
@mwasilew I made the changes requested to use a result file with standardized format, I noticed that increase the running time 1 hour, 7 minutes [1] vs 1 hour, 43 minutes [2]. 

Sometime ago I found some high cpu consumption parsing results using pexpect in lava-dispatcher may be is the currently issue. For now I'm fine with the current times.

[1] https://validation.linaro.org/scheduler/job/2163739
[2] https://validation.linaro.org/scheduler/job/2165199